net/route.go

```diff
@@ -26,3 +26,18 @@ func CheckNetworkFree(subnet *net.IPNet) error {
 func overlaps(n1, n2 *net.IPNet) bool {
 	return n1.Contains(n2.IP) || n2.Contains(n1.IP)
 }
+
+// For a specific address, we only care if it is actually *inside* an
+// existing route, because weave-local traffic never hits IP routing.
+func CheckAddressOverlap(addr net.IP) error {
+	routes, err := netlink.RouteList(nil, netlink.FAMILY_V4)
+	if err != nil {
+		return err
+	}
+	for _, route := range routes {
+		if route.Dst != nil && route.Dst.Contains(addr) {
+			return fmt.Errorf("Address %s would overlap with existing route %s", addr, route.Dst)
```

Comment:
s/would overlap/overlaps

...since we don't treat this as an error, and carry on.